### PR TITLE
Add Non-exception Throwing Specifier to `ranges::ssize`

### LINKED ISSUE #4107: <xutility>: `ranges::_Ssize_fn::operator()` should have a noexcept-specifier
# Describe the bug
`ranges::ssize(a)` is specified to be expression-equivalent to `static_cast<D>(ranges::size(a))` (for some integer-like type `D`), which means applying `noexcept` to these two expressions should produce the same result. But due to a missing *noexcept-specifier*, `noexcept(ranges::ssize(a))` is always `false` on MSVC, which is non-conforming.

This can be fixed by adding a conditional *noexcept-specifier* to `ranges::_Ssize_fn::operator()`. Since `D` is integer-like, the cast should never throw an exception, so the *noexcept-specifier* only needs to mention the `ranges::size` call.

I think this could be a "good first issue".

# Command-line test case
```
D:\test>type test-ssize.cpp
#include <ranges>

int a[10];
static_assert(noexcept(std::ranges::ssize(a)));

D:\test>cl /Zs /std:c++20 test-ssize.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.38.33030 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

test-ssize.cpp
test-ssize.cpp(4): error C2607: static assertion failed
```

# Expected behavior
no error

# STL version

    Microsoft Visual Studio Community 2022
    Version 17.8.0 Preview 2.0

### PR DESCRIPTION
**Pull request by PhiliaTheCat**
Fixes #4107.

Non-exception throwing specifier added in ```stl/inc/xutility:3357```

Besides, some improper fomats are changed in ```stl/inc/xutility```

### CODE REVIEW

**Review by frederick-vs-ja — COMMENTED** [review]
It would be better to use [GitHub's close/fix/resolve syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) (e.g. saying `Fixes #4107`) in the description of this PR (instead of the commit message).

**Comment by PhiliaTheCat** [comment]
@microsoft-github-policy-service agree

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Looks great, thanks! I updated your PR description to properly link to the issue that it resolves, and updated the PR title to mention what ranges function is being changed.

I think this needs only one maintainer approval. We merge PRs simultaneously to the GitHub and MSVC-internal repos, batched up in a semi-manual process to save time. Your PR will be part of the next batch, probably next week.

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by PhiliaTheCat** [comment]
Copy&nbsp;that.&nbsp;(Am&nbsp;i&nbsp;supposed&nbsp;to&nbsp;reoly&nbsp;this?)


---Original---
From: &quot;Stephan T. ***@***.***&gt;
Date: Thu, Dec 7, 2023 10:13 AM
To: ***@***.***&gt;;
Cc: &quot;Cai ***@***.******@***.***&gt;;
Subject: Re: [microsoft/STL] Add Non-exception Throwing Specifier to`ranges::ssize` (PR #4231)




 
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.
 
&mdash;

**Comment by StephanTLavavej** [comment]
Thanks for fixing `ranges::ssize` and congratulations on your first microsoft/STL commit! :tada: :heart_eyes_cat: :rocket:

This will ship in VS 2022 17.10 Preview 1.

**Pull request merged**
